internals/fetchmetrics.py

```diff
@@ -212,7 +211,6 @@ def get_template_data(self, today=None):
                 'WebStatusAlert-1: Failed to get metrics even after 2 days')
           return error_message, 500
 
-    ramcache.flush_all()
```

Comment:
I think we need a `flush_all()` or some other kind of invalidation here.  I should have documented the reason with a comment.

This code calls `FetchAndSaveData()` which calls `_SaveData()` which calls `put()` a bunch of times to add a new entity for each metrics datapoint.     Separately, when a request comes in get get metrics data, the file api/metricsdata.py does a query on those datapoints and caches the result.   If we don't invalidate when we add datapoints, the cached query result will be lacking the new datapoints.

I'm not too worried about doing a `flushall()` in this case, because the fetch metrics cron job only runs once every 6 hours.

Reply:
Added `flushall` and your comment above. I think `flushall` is fine for now until we don't cache lots stuff in this PR, but we need to come up with a better way to invalidate once we start caching features

Reply:
Actually could you elaborate on the cron job?  How is the cron job used?

Reply:
Our metrics data comes from the uma-export app.   We have a cron job configured to run every 6 hours that checks uma-export for new metrics data.  When it finds metrics data, it stores those new datapoints in datastore via `_SaveDatas()`.  That's the code in the fetchmetrics.py file.

When a user views metrics data, that runs code in api/metricsdata.py which can load datapoints from the cache (or on a cache miss it loads from datastore and then puts them in the cache).

If we have a cached value, that value would keep being used until it expires even if we have fresher datapoints in datastore.  E.g., if the cache was written at 11am, but then the cron job ran at 12 noon, we would still be displaying the stale data until 11am the next day, meaning that it was stale by up to 23 hours in that example.

I suppose there are actually two solution approaches for this:
1. We could invalidate the cache somehow, such as calling `flush_all()`.  That would cause users to see new metrics data as soon as we get it.  But, if we flush everything rather than flushing just the metrics data then users viewing other parts of the site will still see some cache misses.
2. Some staleness in metrics data could be acceptable.  23 hours of staleness is probably too much, but if we reduced the expiration time to 2, 3, or 6 hours, users probably would not notice.   A shorter expiration time for this data means that there will be more cache misses for metrics but it would not cause cache misses for other kinds of cached data.

Reply:
That makes sense! Thanks!